bigcache_bench_test.go

```diff
@@ -69,6 +69,89 @@ func BenchmarkReadFromCache(b *testing.B) {
 	}
 }
 
+func BenchmarkReadFromCacheManySingle(b *testing.B) {
+    for _, shards := range []int{1, 512, 1024, 8192} {
+		b.Run(fmt.Sprintf("%d-shards", shards), func(b *testing.B) {
+			cache, _ := New(context.Background(), Config{
+				Shards:             shards,
+				LifeWindow:         1000 * time.Second,
+				MaxEntriesInWindow: max(b.N, 100),
+				MaxEntrySize:       500,
+			})
+
+            keys := make([]string,b.N)
+			for i := 0; i < b.N; i++ {
+                keys[i] = fmt.Sprintf("key-%d", i)
+				cache.Set(keys[i], message)
+			}
+
+            b.ReportAllocs()
+            b.ResetTimer()
+            for _,key := range keys {
+                cache.Get(key) 
+            }
+
+		})
+	}
+}
+
+func BenchmarkReadFromCacheManyMulti(b *testing.B) {
+	for _, shards := range []int{1, 512, 1024, 8192} {
+        b.Run(fmt.Sprintf("%d-shards", shards), func(b *testing.B) {
+            cache, _ := New(context.Background(), Config{
+                Shards:             shards,
+                LifeWindow:         1000 * time.Second,
+                MaxEntriesInWindow: max(b.N, 100),
+                MaxEntrySize:       500,
+            })
+            keys := make([]string,b.N)
+            for i := 0; i < b.N; i++ {
+                keys[i] = fmt.Sprintf("key-%d", i)
+                cache.Set(keys[i], message)
+            }
+
+            b.ReportAllocs()
+            b.ResetTimer()
```

Comment:
why reset here?

Reply:
My idea was to only benchmark the Get method, do you prefer it being before setting the values?